src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -1479,16 +1551,38 @@ void ICanSaveModel.Save(ModelSaveContext ctx)
             for (int index = 0; index < memberInfos.Length; index++)
             {
                 var memberInfo = memberInfos[index];
-                var mappingAttr = memberInfo.GetCustomAttribute<LoadColumnAttribute>();
 
-                if (mappingAttr == null)
-                    throw host.Except($"{(memberInfo is FieldInfo ? "Field" : "Property")} '{memberInfo.Name}' is missing the {nameof(LoadColumnAttribute)} attribute");
+                var ignoreLoadColumnAttr = memberInfo.GetCustomAttribute<LoadColumnIgnoreAttribute>();
+                if (ignoreLoadColumnAttr != null && ignoreLoadColumnAttr.Ignore)
+                    continue;
+
+                var mappingLoadColumnAttr = memberInfo.GetCustomAttribute<LoadColumnAttribute>();
+                var mappingLoadColumnNameAttr = memberInfo.GetCustomAttribute<LoadColumnNameAttribute>();
+                var memberInfoType = memberInfo is FieldInfo ? "Field" : "Property";
+
+                if (!hasHeader && mappingLoadColumnAttr == null)
+                    throw host.Except(
+                        $"You should set {nameof(hasHeader)} to true in order to use {nameof(LoadColumnNameAttribute)}. " +
+                        $"To automatically map the name of {memberInfoType} to {nameof(LoadColumnNameAttribute)}, set {nameof(autoMapLoadColumns)} to true. " +
+                        $"To specify a fixed column index, use {nameof(LoadColumnAttribute)}. " +
+                        $"Invalid {memberInfoType} name: {memberInfo.Name}.");
```

Comment:
This should go in the documentation. Can you please shorten the exception messages (here and below) to be focused on the specific exception?